---
title: PrÃ©parer l'environnement logiciel
author: Pierre Poulain
license: Creative Commons Attribution-ShareAlike (CC BY-SA 4.0)
---

# PrÃ©parer l'environnement logiciel ðŸ§°

On suppose que conda et mamba sont installÃ©s.

## CrÃ©er l'environnement conda

CrÃ©ez un environnement conda :

```bash
conda create -n rnaseq-scere -y
```

Activez ensuite cet environnement :

```bash
conda create -n rnaseq-scere -y
```

## Installer les logiciels nÃ©cessaires

Nous allons utiliser [SRA Toolkit](https://github.com/ncbi/sra-tools) pour tÃ©lÃ©charger les donnÃ©es brutes de sÃ©quenÃ§age.

L'article indique :

> Raw FASTQ files were aligned to the respective yeast genomes using STAR [78]. Aligned reads were assembled into transcripts, quantified, and normalized using Cufflinks2 [79]. Samples from each yeast time series were normalized together using the CuffNorm feature.

Le fichier *S1 Supporting Information Methods* fournit des prÃ©cisions supplÃ©mentaires :

> Raw FASTQ files from each experiment were aligned to the respective yeast reference genome using STAR [1].

> Reads mapping uniquely to annotated gene features were quantified using HTSeq-count [3].

> Transcript quantification of annotated yeast genes was performed using alignment files output from STAR and Cufflinks2 [4]. Time point samples from the respective yeasts were then normalized together using the CuffNorm feature.

En rÃ©sumÃ©, nous avons besoin d'installer les outils : `STAR`, `HTSeq-count` et `Cufflinks`. Aucune version de logiciel n'Ã©tant spÃ©cifiÃ©e, nous allons installer la derniÃ¨re version disponible.

Nous installerons Ã©galement [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) pour contrÃ´ler la qualitÃ© des *reads*.

Dans l'environnement conda `rnaseq-scere`, installez tous les logiciels nÃ©cessaires :

```bash
mamba install -c conda-forge -c bioconda sra-tools fastqc star htseq cufflinks -y
```

VÃ©rifiez alors les diffÃ©rentes versions des logiciels :

```bash
$ fasterq-dump --version

"fasterq-dump" version 2.11.0
```

```bash
$ fastqc --versio
FastQC v0.11.9
```

```bash
 STAR --version
2.7.10a
```

```bash
$ htseq-count --version
0.13.5
```

```bash
$ cufflinks 2>&1 | head -n 1
cufflinks v2.2.1
```